src/XML/src/FMI3/fmi3_xml_model_structure.c

```diff
@@ -300,107 +298,121 @@ static int fmi3_xml_parse_dependencies(fmi3_xml_parser_context_t *context,
+        if (*numDepInd != *numDepKind) {
             fmi3_xml_parse_error(context, "XML element '%s': different number of items (%u and %u) in the lists for 'dependencies' and 'dependenciesKind'",
-                                 fmi3_xml_elmid_to_name(elmID), numDepInd, numDepKind);
-            ms->isValidFlag = 0;
-            return 0;
+                    fmi3_xml_elmid_to_name(elmID), *numDepInd, *numDepKind);
+            return -1;
         }
-        jm_vector_push_back(size_t)(&deps->startIndex, totNumDep + numDepInd);
+        jm_vector_push_back(size_t)(&deps->startIndex, totNumDep + *numDepInd);
         jm_vector_push_back(char)(&deps->dependencyOnAll, 0);
     }
     else if (listInd) {
         /* only Dependencies are present, set all kinds to dependent */
         /* this includes the case of empty dependencies */
         char kind = fmi3_dependencies_kind_dependent;
-        if (jm_vector_reserve(char)(&deps->dependenciesKind, totNumDep + numDepInd) < totNumDep + numDepInd) {
+        if (jm_vector_reserve(char)(&deps->dependenciesKind, totNumDep + *numDepInd) < totNumDep + *numDepInd) {
             fmi3_xml_parse_fatal(context, "Could not allocate memory");
             return -1;
         }
-        for (;numDepKind < numDepInd; numDepKind++) {
+        for (size_t k = 0; k < *numDepInd; k++) {
             jm_vector_push_back(char)(&deps->dependenciesKind, kind);
         }
-        jm_vector_push_back(size_t)(&deps->startIndex, totNumDep + numDepInd);
+        jm_vector_push_back(size_t)(&deps->startIndex, totNumDep + *numDepInd);
         jm_vector_push_back(char)(&deps->dependencyOnAll, 0); // not dependent on all
     }
     else if (listKind) {
         fmi3_xml_parse_error(context, "XML element '%s': if `dependenciesKind` attribute is present then the `dependencies` attribute must also be present.",
             fmi3_xml_elmid_to_name(elmID));
-        ms->isValidFlag = 0;
-        return 0;
-    }
-    else {
-        /* Dependencies attribute is missing*/
+        return -1;
+    } else {
+        /* Dependencies attribute is missing, set default dependencies */
         jm_vector_push_back(size_t)(&deps->startIndex, totNumDep);
         jm_vector_push_back(char)(&deps->dependencyOnAll, 1);
     }
     return 0;
 }
 
+/* Wrapper for #fmi3_xml_parse_dependencies, handling failed states */
+static void fmi3_xml_parse_dependencies_error_wrapper(fmi3_xml_parser_context_t* context,
+        fmi3_xml_elm_enu_t elmID, fmi3_xml_dependencies_t* deps) {
+    // number of added dependecies(Kind) entries, these are needed to clean dependencies in case of failure to parse
+    size_t numDepInd = 0;
+    size_t numDepKind = 0;
+    size_t totNumDep = jm_vector_get_size(size_t)(&deps->dependenciesVRs); // original size, before parsing dependencies
+
+    if (fmi3_xml_parse_dependencies(context, elmID, deps, &numDepInd, &numDepKind)) {
+        // Dependecy parsing failed, cleanup of already parsed entries
```

Comment:
OPTIONAL: Could add a more explicit message here that the previous error in parsing the dependencies results in using the default dependencies. Not sure if this is helpful or just extra bloat.

Reply:
I think we are fine leaving it as it is for now.